### PR TITLE
48-reset_index_inplace

### LINKED ISSUE #48: Feature thought - chainable, in place reset_index
As far as I'm aware, an `inplace=True` `reset_index()` does not return a `DataFrame`.

### PR DESCRIPTION
**Pull request by zbarry**
This PR resolves #48 .

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from <your_username>:master, but rather from <your_username>:<branch_name>.
2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [x] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checkin
    - running the test suite
    - docs build

## Code Changes

If you are adding code changes, please ensure the following:

- [x] Ensure that you have added tests.
- [x] Run all tests (`$ pytest .`) locally on your machine.
    - [x] Check to ensure that test coverage covers the lines of code that you have added.
    - [x] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [x] Build the docs locally.
- [x] View the docs to check that it renders correctly.

# PR Description

Please describe the changes proposed in the pull request:

reset_index_inplace

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by zbarry** [comment]
ready.

File: janitor/functions.py [hunk 1]
```diff
@@ -1575,6 +1571,50 @@ def collapse_levels(df: pd.DataFrame, sep: str = "_"):
...
+
+    instead, being called simply as:
+
+    .. code-block:: python
+
+        df = (
+            pd.DataFrame(...)
+            .operation1(...)
+            .operation2(...)
+            .reset_index_inplace()
```
Comment (ericmjl) [inline janitor/functions.py]: I'd suggest swapping `.operation2()` and `.reset_index_inplace()`, in the "before" and "after" just to emphasize that `reset_index_inplace()` is the method-chaining equivalent of `reset_index(inplace=True)`.

File: janitor/functions.py [hunk 2]
```diff
@@ -546,16 +546,12 @@ def fill_empty(df, columns, value):
         for col in columns:
             assert (
                 col in df.columns
-            ), "{col} missing from dataframe columns!".format(
-                col=col
-            )
+            ), "{col} missing from dataframe columns!".format(col=col)
```
Comment (ericmjl) [inline janitor/functions.py]: We might be having black formatting clashes here. Can I check, are you using `black -l 79`? Or is there something else that's causing this? I noticed that between PRs, this flip-flops in formatting.

Comment (zbarry) [reply janitor/functions.py]: Just purely running `make check` which runs `black -l 79 .`

Comment (ericmjl) [reply janitor/functions.py]: huh, dunno then. No worries.

Comment (zbarry) [reply janitor/functions.py]: _shrug_

File: janitor/functions.py [hunk 3]
```diff
@@ -1575,6 +1571,52 @@ def collapse_levels(df: pd.DataFrame, sep: str = "_"):
     return df
 
 
+@pf.register_dataframe_method
+def reset_index_inplace(df: pd.DataFrame, *args, **kwargs):
+    """
+    In Pandas, `reset_index()`, when used in place, does not return a
```
Comment (ericmjl) [inline janitor/functions.py]: I think it's also important to emphasize why this exists over doing method-chained `.reset_index()`. Memory usage, IIRC?

Please let me know what you think about it.

```suggestion
    Returns the dataframe with an inplace resetting of the index.
    
    Compared to non-inplace resetting, this avoids data copying, thus
    providing a potential speedup.
 
    In Pandas, `reset_index()`, when used in place, does not return a
```

**Comment by ericmjl** [comment]
@zbarry please let me know what you think about the final suggested change.

File: janitor/functions.py (hunk 3 above)
Comment (zbarry) [reply janitor/functions.py]: Both memory and compute time, yeah. Looks good.

**Comment by zbarry** [comment]
@ericmjl anything else ya want?

**Comment by ericmjl** [comment]
I think it's all good. LGTM!

**Comment by zbarry** [comment]
👍

**Pull request merged**